### ISSUE TITLE
Collection of issues

### ISSUE THREAD

**Issue by Peyje** [kc0bfv/autophugo#1]
While I quite like this template, I did encounter a few bugs. Instead of creating a new issue for every single one, I will just list them here (some might also just be user error tbh). Feel free to create separate issues for them, if so desired.

- [ ]  Hugo throws the warning `Module "autophugo" is not compatible with this Hugo version` every time I build the site. I am using the latest autophugo version and the newest hugo version (0.57.2) available for Ubuntu. However, the built site looks as expected, so I cannot tell if this affects anything.
- [ ]  Resizing and `hugo serve`: After changing the thumb-width or full-width in the config.yaml already resized pictures are not discarded and resized again to the new values. _Workaround_: Delete the folder `resources`, then all pictures are resized again.
- [ ] Resizing and `hugo`: Didn't work for me at all. Both thumbnail and full picture are created in the correct folder, however both are of the original size. _Workaround_: I manually resized them by using `mogrify -resize 480 *480*.jpg && mogrify -resize 960 *960*.jpg`.
- [ ] Yet another resize issue: The resized pictures still got `480` or `960` in their names, even if the thumb-width/full-width has been changed in the `config.toml`.
- [ ] When downloading a picture, the filename is the name as stored in the public folder. Could this file rather have the original name as in the asset folder?
- [ ] When using the site, the arrow keys move vertically (e.g. 1->3->5->2->4) through the pictures.
- [ ] The Installation part is missing in the Readme.
- [ ] I found the default width values of 480 and 960 too low for a normal 1080p display, I instead used 960 and 1920 respectively. That might just be a preference, but 480 looks too pixelated for me.

Still, this is my favorite gallery template, thank you for your good work! I appreciate any reply and would be glad to help if needed.

**Comment by kc0bfv** [comment]
Hey!  Awesome.
- regarding version, I may have set the minimum version too high.  I think I based it on one of the resources features I used.  That's a pretty old version of Hugo though - I'm a Debian guy and my distro has a similarly old version.  If you use the snap install you get a much newer version - 0.69.2.  https://gohugo.io/getting-started/installing/#snap-package
- regarding resizing - thanks, I'll look at that.  I think my version kept them around too, but would've had new filenames and URLs for the differently-sized photos.  So, the right photos would get used, but the old ones would be there bloating up everything.  Definitely don't want that...
- I think this resizing issue is due to your version.
- I'll check this out next time I build.  I think mine had some automatically created filenames with the correct sizes.
- Filenames via download - yeah, that's a great idea.  There's probably a way to do this.  I also think right now it doesn't have a way to link to an "original" image, like a full-size given somewhere else...  That could be good.
- The arrow keys sounds like a real accessibility issue.  I'd have to look into how to fix that one.
- Installation - true.  I can link to Hugo's documentation on theme installation.
- It does look pixelated by default.  Increasing the default could be good.

I think several of these things will work differently if you use a newer Hugo - I can't recommend the snap installs more highly.  Apparently - snap is installed by default on recent Ubuntus.  https://snapcraft.io/docs/installing-snap-on-ubuntu

Most of this was shamelessly stolen from the old Phugo.  I loved it too.  The features in Hugo since Phugo was created are pretty awesome though, and make building a purdy gallery almost as easy as dropping photos in a folder.

**Comment by kc0bfv** [comment]
Oh - if you get a chance to tackle these issues and submit a pull request, I'd love that.  If not, I'll probably look into them within a week or so.

**Comment by Peyje** [comment]
I do not currently have the capacities to work on it, but will probably have it in a few weeks of time.
I will then try out the snap version and see if I can resolve some of these issues, if still open, absolutely!

**Comment by kc0bfv** [comment]
- There were some changes to hugo at around .58 in resources and image processing, and I think your experience shows at least .58 is needed.
- My resources stick around too, but actually that's not terrible behavior...  Hugo doesn't put them into the deployment folder unless they're used.  So those extra resources won't bloat up deployment.  They will just be on the developer's computer, which is nice if the developer is trying multiple size/quality options.  They won't have to wait for Hugo to delete/recreate every time if they go back to one combo they've tried previously.  The developer can delete their resources directory once when they settle on a quality/size combo.  I added a note about this to the README.
- Resizing works well for me.
- The names come out correct too - the naming is actually a Hugo feature, I don't do anything custom.
- Downloading now has the original filename - it does not have the original size...  I considered this - maybe in the future I'd just add a download URL option to the images in the section front matter...
- I can't get arrow keys to work at all here, honestly.  Reading about some accessibility I really need three things - I need to have the images specified in the source in the order we want to tab through them, I need to make sure they're picked as things that can be clicked on (add a tabindex to each image), and I need to make sure that they can actually be clicked on.  Not sure what's up with number three.  I just added a tabindex 0 to each image anchor tag.  I don't have a way to reorder the images in the source, and still have them flow in this two column format...  I'll have to think about this.  This will become its own bug.
- The installation section - I see what you mean now.  I forgot about the table of contents there - there were several things missing.  I just nixed the TOC, and added a link to Hugo's theme instructions.
- I added some quality options when I was going the code.  I realized the quality might also have big effects that people want to tune.
